src/libmongoc/src/mongoc/mongoc-collection.h

```diff
@@ -351,6 +351,30 @@ mongoc_collection_estimated_document_count (
    bson_t *reply,
    bson_error_t *error);
 
+
+
+
+typedef struct _mongoc_index_model_t mongoc_index_model_t;
+typedef struct _mongoc_index_options_t mongoc_index_options_t;
+
+MONGOC_EXPORT (bool)
+mongoc_collection_create_indexes (
+   mongoc_collection_t *collection,
+   const mongoc_index_model_t **indexes,
+   size_t num_indexes,
+   const bson_t *create_indexes_opts,
+   char **index_names, /* out parameter, populated with index names */
+   int *num_indexes_created, /* out parameter contained length of index_names */
```

Comment:
Added out parameters for index_names created and length of index_names because https://github.com/mongodb/specifications/blob/master/source/index-management.rst indicates that `create_indexes` should return names of indexes created.

Reply:
I would rename this to `num_index_names` for consistency.

If you look at the [`createIndexes` output](https://docs.mongodb.com/manual/reference/command/createIndexes/#output), you'll see that it doesn't actually report which indexes were created. It merely says how many existed before and after the command was run. The return value here is derived entirely from the input argument and has no relation to the server response, apart from whether or not the command succeeded or failed.

IIRC, the index management spec doesn't address this detail, although I agree it'd be helpful to clarify that if you want to take a stab at creating a DRIVERS ticket to propose a change (Kevin, Clyde, or Alex can advise on workflow).